src/core/infer_request.h

```diff
@@ -442,6 +473,11 @@ class InferenceRequest {
   InferenceResponseFactory response_factory_;
 
   std::vector<std::function<void()>> release_callbacks_;
+
+  // timestamps that is request specific
+  uint64_t request_start_ns_;
+  uint64_t queue_start_ns_;
+  std::vector<std::shared_ptr<StatsAggregator>> stats_collectors_;
```

Comment:
secondary_stats_aggregator_
Just 1, not a vector. I think it can be a raw pointer since the ensembler lifetime must be longer than individual requests (and this aligns with what it will be in the future when exposed in TRITONSERVER)